src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ExportProvider.cs

```diff
@@ -10,148 +10,57 @@
 
 namespace Internal.Cryptography.Pal
 {
-    internal sealed class ExportProvider : IExportPal
+    internal sealed class ExportProvider : UnixExportProvider
     {
-        private static readonly SafeEvpPKeyHandle InvalidPKeyHandle = new SafeEvpPKeyHandle(IntPtr.Zero, false);
-
-        private ICertificatePalCore _singleCertPal;
-        private X509Certificate2Collection _certs;
-
         internal ExportProvider(ICertificatePalCore singleCertPal)
+            : base(singleCertPal)
         {
-            _singleCertPal = singleCertPal;
         }
 
         internal ExportProvider(X509Certificate2Collection certs)
+            : base(certs)
         {
-            _certs = certs;
         }
 
-        public void Dispose()
+        protected override byte[] ExportPkcs8(
+            ICertificatePalCore certificatePal,
+            ReadOnlySpan<char> password)
         {
-            // Don't dispose any of the resources, they're still owned by the caller.
-            _singleCertPal = null;
-            _certs = null;
-        }
+            AsymmetricAlgorithm alg = null;
+            SafeEvpPKeyHandle privateKey = ((OpenSslX509CertificateReader)certificatePal).PrivateKeyHandle;
 
-        public byte[] Export(X509ContentType contentType, SafePasswordHandle password)
-        {
-            Debug.Assert(password != null);
-            switch (contentType)
+            try
             {
-                case X509ContentType.Cert:
-                    return ExportX509Der();
-                case X509ContentType.Pfx:
-                    return ExportPfx(password);
-                case X509ContentType.Pkcs7:
-                    return ExportPkcs7();
-                case X509ContentType.SerializedCert:
-                case X509ContentType.SerializedStore:
-                    throw new PlatformNotSupportedException(SR.Cryptography_Unix_X509_SerializedExport);
-                default:
-                    throw new CryptographicException(SR.Cryptography_X509_InvalidContentType);
+                alg = new RSAOpenSsl(privateKey);
```

Comment:
is there a way to do so without all the try catches?

Reply:
* Import or redefine the P/Invokes that this ctor would use
* Import the SafeHandle types it needs
* Repeat the work done in the ctors, without the throw.

So right now, I'd say "no", but perhaps this suggests we want something like `public static bool RSAOpenSsl.TryCreate(SafeEvpPkeyHandle, out RSAOpenSsl)` which leaves the work in the other assembly and avoids the try/catch.